### ISSUE TITLE
Getting InvalidOperationException when resolving Shiny.BluetoothLE.Central.CentralManager

### ISSUE THREAD

**Issue by pdecostervgls** [shinyorg/shiny#409]
ONLY active OSS contributors OR people who buy me a coffee can ask questions here.  If you don't do either of these things - DO NOT FILE HERE OR I WILL SIMPLY REMOVE YOUR ACCOUNT :)

Give as much details as humanly possible if you want any sort of answer!

Enter Question Below (don't delete this line)

Hey Allan,

So just to prevent this getting closed immediately, I bought you a coffee, I am an OSS contributor (Uno Platform and Plugin.BluetoothLE) and I'm trying to get my company to fund your project.

So I am working on an Uno Platform app for a while now, first I used Plugin.BluetoothLE and recently switched over to Shiny.

I got Shiny to work on UWP and iOS, but with Android I have being having a heap of issues.
The biggest issue is the following exception:

```
System.InvalidOperationException: 'A suitable constructor for type 'Shiny.BluetoothLE.Central.CentralManager' could not be located. Ensure the type is concrete and services are registered for all parameters of a public constructor.'
```

Below is part of my Main.cs file:
```
    public class Application : Windows.UI.Xaml.NativeApplication
    {
        public Application(IntPtr javaReference, JniHandleOwnership transfer)
            : base(() => new App(), javaReference, transfer)
        {
            ConfigureUniversalImageLoader();
        }

        public override void OnCreate()
        {
            base.OnCreate();
            Shiny.AndroidShinyHost.Init(
                this,
                new MotionMountShinyStartup()
            );
            CrossCurrentActivity.Current.Init(this);
            Acr.UserDialogs.UserDialogs.Init(this);
        }
        ...
```
I wasn't able to inherit the Application from ShinyAndroidApplication as the documentation suggests, when I did that, the application crashed on an NullReferenceException

The line that the application throws the exception is: 
```
ICentralManager manager = ShinyHost.Resolve<ICentralManager>();
```

I hope that's enough information to start from, please advise.

**Comment by sgermosen** [comment]
Change IcentralManager to IbleManager or downgrade the package to 1.1

**Comment by aritchie** [comment]
BLE is beta and the class names have changed from ICentralManager to IBleManager.  There will be a few more changes before it is released, so upgrade at your own risk.

**Comment by pdecostervgls** [comment]
So I did as you asked, and updated my repo with the dev changes. Changed the name to `IBleManager`, but I get the same error:
```
System.InvalidOperationException: 'A suitable constructor for type 'Shiny.BluetoothLE.BleManager' could not be located. Ensure the type is concrete and services are registered for all parameters of a public constructor.'
```
My question was mostly about whether I can use the Init method like I do, or if I really have to make the `Application` inherit from `ShinyAndroidApplication`?

**Comment by aritchie** [comment]
You can inherit from Android application or the ShinyAndroidApplication.  It doesn't matter in this case.  What you're showing is the Uno application though, in which case, I'm not sure where that executes.

**Comment by aritchie** [comment]
If you submit a sample with this problem, I can figure out what the issue is.

**Comment by pdecostervgls** [comment]
> If you submit a sample with this problem, I can figure out what the issue is.

I'll get right on that. Thanks for the help!

**Comment by pdecostervgls** [comment]
@aritchie I wasn't able to reproduce the issue in an seperate project. The other project has it working.
I am currently investigation what the differences are between my main project and and the reproduction one.

One of the things i've noticed is that the main project reports that the `BleManager` class doesn't have any constructors. The code shows that it does indeed have constructors.

The code I used is:
```
IServiceCollection services = Shiny.ShinyHost.Services;
var bleManagerService = services[2]; // Take the IBleManager service, which is the 3rd in the list
var constructors = bleManagerService.ImplementationType.GetConstructors();
```

Where constructors shows having a count of 1 in the repro project, and 0 on my main project.

I am at a loss what steps I can take to investigate this discrepancy further, I tried to read the exports of the Shiny.BluetoothLE.dll file in my android projects "obj" folder, but it shows no exported functions. The object browser in VS also shows that `BleManager` has an constructor, in both projects.
